common/components/ui/Dropdown.tsx

```diff
@@ -79,7 +79,9 @@ export default class DropdownComponent<T> extends Component<Props<T>, {}> {
   };
 
   private onChange = (value: any) => {
-    this.props.onChange(value);
+    if (this.props.onChange) {
+      this.props.onChange(value);
+    }
```

Comment:
Hm, the usecase i see for disabling this drop down is for when the page is loaded with a `readOnly` query selector. Perhaps it would be better to just replace the dropdown with a `span` with the token / ether type when readOnly is selected? It would make it obvious to the user that it's not clickable.